source/simulator/nullspace.cc

```diff
@@ -385,33 +385,13 @@ namespace aspect
           const std::vector<Point<dim> > &q_points = fe.get_quadrature_points();
 
           // get the density at each quadrature point if necessary
-          MaterialModel::MaterialModelInputs<dim> in(n_q_points,
-                                                     parameters.n_compositional_fields);
+          MaterialModel::MaterialModelInputs<dim> in(fe,
+                                                     &cell,
+                                                     this->introspection,
+                                                     relevant_dst);
           MaterialModel::MaterialModelOutputs<dim> out(n_q_points,
-                                                       parameters.n_compositional_fields);
-
-          //Get the velocity at each quadrature point
-          fe[introspection.extractors.velocities].get_function_values (relevant_dst, in.velocity);
-
-          if ( ! use_constant_density)
-            {
-              fe[introspection.extractors.pressure].get_function_values (relevant_dst, in.pressure);
-              fe[introspection.extractors.temperature].get_function_values (relevant_dst, in.temperature);
-              fe[introspection.extractors.pressure].get_function_gradients (relevant_dst, in.pressure_gradient);
-              for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
-                fe[introspection.extractors.compositional_fields[c]].get_function_values(relevant_dst,
-                                                                                         composition_values[c]);
+                                                       introspection.n_compositional_fields);
 
-              for (unsigned int i=0; i<n_q_points; ++i)
-                {
-                  in.position[i] = fe.quadrature_point(i);
-                  for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
-                    in.composition[i][c] = composition_values[c][i];
-                }
-              in.cell = &cell;
-
-              material_model->evaluate(in, out);
```

Comment:
nullspace.cc has been reverted back to it's original form. I have tested this new call in source/simulator/lateral_averaging.cc instead. The test passed. I have updated the commit message accordingly.